### PR TITLE
Update storage assets documentation

### LINKED ISSUE #595: Update documentation for seasonal and non-seasonal storage
### What and Why

- [x] Change the seasonal storage modelling concepts to have a more general section with Storage Modelling. Add comments on modelling non-seasonal storage, too. Also, modelling options for both types of storage.
- [x] Update the figures to have the correct variable names (according to mathematical formulation)
- [x] Don't include a tutorial on hydrothermal dispatch. See comment on #567 why, at this date, we can't model that case study in Tulipa.

### Possible Drawbacks

_No response_

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by datejada**
# Pull request details

## Describe the changes made in this pull request

Update storage asset documentation to show how to model seasonal and non-seasonal storage.

## List of related issues or pull requests

Closes #595 

## Collaboration confirmation

As a contributor I confirm

-   [x] I read and followed the instructions in README.dev.md
-   [x] The documentation is up to date with the changes introduced in this Pull Request (or NA)
-   [x] Tests are passing
-   [x] Lint is passing

### CODE REVIEW

File: docs/src/concepts.md [hunk 1]
```diff
@@ -20,11 +20,11 @@ _TulipaEnergyModel.jl_ incorporates two fundamental concepts that serve as the f
...
--   **Inter-temporal Constraints**: These constraints combine the information of the representative periods and create limitations between them to recover chronological information across the full timeframe. The inter-temporal constraints help to characterize the long-term operational dynamics of the assets. So far, the model uses this type of constraint to model seasonal storage, but future developments will include, for example, maximum or minimum production/consumption for a year (or any timeframe).
+-   **Inter-temporal Constraints**: These constraints combine the information of the representative periods and create limitations between them to recover chronological information across the whole timeframe. The inter-temporal constraints help to characterize the long-term operational dynamics of the assets. So far, the model uses this type of constraint to model seasonal storage. Still, future developments will include, for example, maximum or minimum production/consumption for a year (or any timeframe).
```
Comment (clizbe) [inline docs/src/concepts.md:23]: I actually prefer the "storage, but" wording.

**Review by clizbe — CHANGES_REQUESTED** [review]
Looks pretty good! I think we should just be careful mixing Concept and How-To. The rule of thumb is you can read Concept in the bath, where as you would read How-To if you're actively implementing something.

File: docs/src/concepts.md (hunk 1 above)
Comment (datejada) [reply docs/src/concepts.md:23]: I added, this example at the end of the sentence to guide what we meant as a use case...it is trying to be general as an application:

> The inter-temporal constraints help to characterize the long-term operational dynamics of the assets (e.g., seasonality)

File: docs/src/concepts.md [hunk 2]
```diff
@@ -330,9 +330,23 @@ The level of reduction and approximation error will depend on the case study. So
...
+Energy storage systems can be broadly classified into two categories: seasonal and non-seasonal storage. Seasonal storage refers to assets that can store energy for more extended periods, usually spanning months or even years. Examples of such assets include hydro reservoirs, hydrogen storage in salt caverns, or empty gas fields. On the other hand, non-seasonal storage refers to assets that can store energy only for a few hours, such as batteries or small pumped-hydro storage units.
+
+Both storage categories can be represented in _TulipaEnergyModel.jl_ using the representative periods approach:
+
+-   _Non-seasonal storage_: When the storage capacity of an asset is lower than the total length of representative periods, like in the case of a battery with a storage capacity of 4 hours and representative periods of 24-hour timesteps, intra-temporal constraints should be applied. To apply these constraints, you need to set the input parameter `is_seasonal` to `false` in the [`assets-data.csv`](@ref schemas).
```
[the remark this answers is not in the document]
Comment (datejada) [reply docs/src/concepts.md:339]: I agree, and done in the new changes

File: docs/src/concepts.md [hunk 3]
```diff
@@ -330,9 +330,23 @@ The level of reduction and approximation error will depend on the case study. So
...
+-   _Seasonal storage_: When the storage capacity of an asset is greater than the total length of representative periods, like in the case of a hydroplant with a storage capacity of a month and representative periods of 24-hour timesteps, inter-temporal constraints should be applied. To apply these constraints, you need to set the input parameter `is_seasonal` to `true` in the [`assets-data.csv`](@ref schemas).
```
[the remark this answers is not in the document]
Comment (datejada) [reply docs/src/concepts.md:340]: same here too

File: docs/src/concepts.md [hunk 4]
```diff
@@ -330,9 +330,23 @@ The level of reduction and approximation error will depend on the case study. So
...
+
+The equations of both intra- and inter-temporal constraints for energy storage is available in the [`mathematical formulation`](@ref formulation). An example is shown in the following section to explain these concepts.
+
+> **Note:**
+> If the input data covers only one representative period for the entire year, for example, with 8760-hour timesteps, and you have a monthly hydropower plant, then you should set the `is_seasonal` parameter for that asset to `false`. This is because the length of the representative period is greater than the storage capacity of the storage asset.
```
[the remark this answers is not in the document]
Comment (datejada) [reply docs/src/concepts.md:345]: It makes more sense in the how-to, so moved there

**Review by clizbe — APPROVED** [review]
Looks great! Thanks for breaking out the technical bit into a new section.

**Pull request merged**